packages/std/src/math/uint256.rs

```diff
@@ -0,0 +1,560 @@
+use schemars::JsonSchema;
+use serde::{de, ser, Deserialize, Deserializer, Serialize};
+use std::convert::TryFrom;
+use std::fmt;
+use std::iter::Sum;
+use std::ops::{self, Shr};
+
+use crate::errors::{DivideByZeroError, OverflowError, OverflowOperation, StdError};
+
+/// This module is purely a workaround that lets us ignore lints for all the code
+/// the `construct_uint!` macro generates.
+#[allow(clippy::all)]
+mod uints {
+    uint::construct_uint! {
+        pub struct U256(4);
+    }
+}
+
+/// Used internally - we don't want to leak this type since we might change
+/// the implementation in the future.
+use uints::U256;
+
+/// An implementation of u256 that is using strings for JSON encoding/decoding,
+/// such that the full u256 range can be used for clients that convert JSON numbers to floats,
+/// like JavaScript and jq.
+///
+/// # Examples
+///
+/// Use `from` to create instances out of primitive uint types or `new` to provide big
+/// endian bytes:
+///
+/// ```
+/// # use cosmwasm_std::Uint256;
+/// let a = Uint256::from(258u128);
+/// let b = Uint256::new([
+///     0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8,
+///     0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8,
+///     0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8,
+///     0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 1u8, 2u8,
+/// ]);
+/// assert_eq!(a, b);
+/// ```
+#[derive(Copy, Clone, Default, Debug, PartialEq, Eq, PartialOrd, Ord, JsonSchema)]
+pub struct Uint256(#[schemars(with = "String")] U256);
+
+impl Uint256 {
+    pub const MAX: Uint256 = Uint256(U256::MAX);
+
+    /// Creates a Uint256(value) from a big endian representation. It's just an alias for
+    /// `from_big_endian`.
+    pub fn new(value: [u8; 32]) -> Self {
+        Self::from_big_endian(value)
+    }
+
+    /// Creates a Uint256(0)
+    pub const fn zero() -> Self {
+        Uint256(U256::zero())
+    }
+
+    pub fn from_big_endian(value: [u8; 32]) -> Self {
```

Comment:
I'd use the same naming as the Rust primitive types: `from_be_bytes` (https://doc.rust-lang.org/std/primitive.u64.html#method.from_be_bytes)